lib/parsers/aac_parser.rb

```diff
@@ -0,0 +1,34 @@
+require_relative 'aac_parser/adts_header_info'
+class FormatParser::AACParser
+  include FormatParser::IOUtils
+
+  AAC_MIME_TYPE = 'audio/aac'
+
+  def likely_match?(filename)
+    filename =~ /\.aac?$/i
+  end
+
+  def call(raw_io)
+    io = FormatParser::IOConstraint.new(raw_io)
+    header = safe_read(io, 9)
+    header_bits = header.unpack('B*').first.split('')
+
+    header_info = FormatParser::AdtsHeaderInfo.parse_adts_header(header_bits)
```

Comment:
Nice separation

Reply:
Thanks. The header had such a specific format that lent itself to a separate implementation, and it also made sense for testing.